### ISSUE TITLE
Build applications without AVX

### ISSUE THREAD

**Issue by rahulchaphalkar** [google/mediapipe#1978]
I have built examples like face_mesh_cpu on windows successfully. Is it possible to build without using AVX 512 / AVX 2 etc. ? I tried passing flags to msvc compiler using the `--copt="-arch:AVX"` flag ([Reference](https://docs.microsoft.com/en-us/cpp/build/reference/arch-x64?view=msvc-160)) but that didn't make a difference.
It seems like if the machine is capable of AVX 512, it uses AVX 512 (similarly AVX2). 
I was hoping to get a build with just SSE 4 for example.

**Comment by jiuqiant** [comment]
Does https://stackoverflow.com/questions/52969630/tensorflow-with-bazel-avx-ignored/53006033 help?

**Comment by rahulchaphalkar** [comment]
It doesn't. I have tried to use `--copt="-arch:AVX"` and `--copt="-arch:AVX2"` etc like the above link suggests, but that didn't seem to do anything different. I'll retry again without the quotes `"` I've used in case its a syntax error, but msvc didn't complain about the flag, so for now I don't think it's that.

**Comment by jiuqiant** [comment]
How about the first answer of https://stackoverflow.com/questions/41293077/how-to-compile-tensorflow-with-sse4-2-and-avx-instructions? To be frank, I don't have a good answer to your question and the only thing I can do is to find the possible solutions online :/

**Comment by rahulchaphalkar** [comment]
This has been resolved, here's a summary -
Mediapipe uses tensorflowlite delegate , which uses Xnnpack library, and xnnpack library has AVX instructions enabled via intrinsics. So, using a compiler flag like`--copt="-arch:AVX"` or others won't have an effect. I was able to disable the `cpuinfo` checks in xnnpack so that it generated only upto SSE 4 instructions.
If anyone is interested, i've put some more details in an issue in xnnpack https://github.com/google/XNNPACK/issues/1453
Closing this issue, thanks.
